docs/cluster-proportional-autoscaler.md

```diff
@@ -0,0 +1,121 @@
+# Horizontal cluster-proportional-autoscaler container
+
+Horizontal cluster-proportional-autoscaler watches over the number of schedulable nodes and cores of the cluster and resizes the number of replicas for the required resource. This functionality may be desirable for applications that need to be autoscaled with the size of the cluster, such as CoreDNS and other services that scale with the number of nodes/pods in the cluster.
```

Comment:
Maybe add a comment on when this will be useful for CoreDNS ? from which number of nodes we need this addon ?

Reply:
To add such recommendations, we need to do testing and gather data. I added a link to the eks best practices guide for this.